ocaml/xapi-guard/lib/server_interface.ml

```diff
@@ -161,27 +180,30 @@ let serve_forever_lwt_callback_vtpm ~cache mutex vtpm _path _ req body =
   (* TODO: some logging *)
   match (Cohttp.Request.meth req, Uri.path uri) with
   | `GET, key when key <> "/" ->
-      let* contents = with_xapi ~cache @@ VTPM.get_contents ~self:vtpm in
+      let* self = get_vtpm_ref () in
+      let* contents = with_xapi ~cache @@ VTPM.get_contents ~self in
       let body = contents |> deserialize |> lookup_key key in
       let headers =
         Cohttp.Header.of_list [("Content-Type", "application/octet-stream")]
       in
       Cohttp_lwt_unix.Server.respond_string ~headers ~status:`OK ~body ()
   | `PUT, key when key <> "/" ->
       let* body = Cohttp_lwt.Body.to_string body in
-      let* contents = with_xapi ~cache @@ VTPM.get_contents ~self:vtpm in
+      let* self = get_vtpm_ref () in
+      let* contents = with_xapi ~cache @@ VTPM.get_contents ~self in
```

Comment:
Same comment here, if the roundtrips become a problem we can add a dedicated API in XAPI to make this a single API call.